api/v1beta2/flowcollector_types.go

```diff
@@ -0,0 +1,901 @@
+}
+
+// `ClientTLS` defines TLS client configuration
+type ClientTLS struct {
+	//+kubebuilder:default:=false
+	// Enable TLS
+	Enable bool `json:"enable,omitempty"`
+
+	//+kubebuilder:default:=false
+	// `insecureSkipVerify` allows skipping client-side verification of the server certificate.
+	// If set to true, the `caCert` field is ignored.
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	// `caCert` defines the reference of the certificate for the Certificate Authority
+	CACert CertificateReference `json:"caCert,omitempty"`
+
+	// `userCert` defines the user certificate reference and is used for mTLS (you can ignore it when using one-way TLS)
+	// +optional
+	UserCert CertificateReference `json:"userCert,omitempty"`
+}
+
+type SASLType string
+
+const (
+	SASLDisabled    SASLType = "DISABLED"
+	SASLPlain       SASLType = "PLAIN"
+	SASLScramSHA512 SASLType = "SCRAM-SHA512"
+)
+
+// `SASLConfig` defines SASL configuration
+type SASLConfig struct {
+	//+kubebuilder:validation:Enum=DISABLED;PLAIN;SCRAM-SHA512
+	//+kubebuilder:default:=DISABLED
+	// Type of SASL authentication to use, or `DISABLED` if SASL is not used
+	Type SASLType `json:"type,omitempty"`
+
+	// Reference to the secret or config map containing the client ID
+	ClientIDReference FileReference `json:"clientIDReference,omitempty"`
+
+	// Reference to the secret or config map containing the client secret
+	ClientSecretReference FileReference `json:"clientSecretReference,omitempty"`
+}
+
+// `DebugConfig` allows tweaking some aspects of the internal configuration of the agent and FLP.
+// They are aimed exclusively for debugging. Users setting these values do it at their own risk.
+type DebugConfig struct {
+	// `env` allows passing custom environment variables to underlying components. Useful for passing
+	// some very concrete performance-tuning options, such as GOGC and GOMAXPROCS, that should not be
+	// publicly exposed as part of the FlowCollector descriptor, as they are only useful
+	// in edge debug or support scenarios.
+	//+optional
+	Env map[string]string `json:"env,omitempty"`
+}
+
+// Add more exporter types below
+type ExporterType string
+
+const (
+	KafkaExporter ExporterType = "KAFKA"
+	IpfixExporter ExporterType = "IPFIX"
+)
+
+// `FlowCollectorExporter` defines an additional exporter to send enriched flows to.
+type FlowCollectorExporter struct {
+	// `type` selects the type of exporters. The available options are `KAFKA` and `IPFIX`. `IPFIX` is unsupported (*).
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="KAFKA";"IPFIX"
+	// +kubebuilder:validation:Required
+	Type ExporterType `json:"type"`
+
+	// Kafka configuration, such as the address and topic, to send enriched flows to.
+	// +optional
+	Kafka FlowCollectorKafka `json:"kafka,omitempty"`
+
+	// IPFIX configuration, such as the IP address and port to send enriched IPFIX flows to. [Unsupported (*)].
+	// +optional
+	IPFIX FlowCollectorIPFIXReceiver `json:"ipfix,omitempty"`
```

Comment:
A `diff` with v1beta1 shows some unintended changes here, here's a fix:

```suggestion
	// `type` selects the type of exporters. The available options are `KAFKA` and `IPFIX`.
	// +unionDiscriminator
	// +kubebuilder:validation:Enum:="KAFKA";"IPFIX"
	// +kubebuilder:validation:Required
	Type ExporterType `json:"type"`

	// Kafka configuration, such as the address and topic, to send enriched flows to.
	// +optional
	Kafka FlowCollectorKafka `json:"kafka,omitempty"`

	// IPFIX configuration, such as the IP address and port to send enriched IPFIX flows to.
	// +optional
	IPFIX FlowCollectorIPFIXReceiver `json:"ipfix,omitempty"`
```

Reply:
good catch !